### PR TITLE
TASK-4121 Add Player.performCommand(command) to execute arbitrary Minecraft commands

### PR DESCRIPTION
**Pull request by inkarkat**
python/mcpi [faf65db..662ba98](https://github.com/ResilientGroup/mcpi/compare/faf65db...662ba98):
* TASK-4121 Add Minecraft.performCommand(command) to execute arbitrary Minecraft commands on the console
* TASK-4121 Add Player.performCommand(command) to execute arbitrary Minecraft commands
* Refactoring: Extract Connection.sendReceiveBool()

---

- [X] The main commit(s) reference the Fibery ticket via a `TASK-NNNN` prefix in the commit message subject
- [X] Include a human-readable description of what the pull request is trying to accomplish
- [X] The CI build passes
---

- [ ] New automated tests have been added
- [ ] The changes are already covered by automated tests and have been adapted
- [ ] These manual test cases cover this change:
- [X] Steps for the reviewer(s) on how they can manually QA the changes:
  - [X] Execute `mc.player.performCommand("help")`
  - [X] Add commands to the whitelist [^1] and execute an allowed (`mc.performCommand("help")`) and disallowed console command (`mc.performCommand("stop")`)
- [ ] This is a minor internal change; basic CI/CD coverage is enough
- [ ] This is an incomplete feature hidden behind feature flag:
- [ ] This is proof-of-concept / experimental code not for production / marked `@Deprecated`
- [ ] No (significant) changes to production code
---

- [X] Classes and public methods have documentation (that doesn't just repeat the technical subject in English)
- [ ] Logging is implemented to monitor feature usage and troubleshoot problems in production
- [ ] These ReWiki pages are affected by this change and will be adapted:

---

- [ ] The change is unrelated to Minecraft
- [X] Works in Java edition
- [X] Works in Bedrock edition

[^1]: plugins/JuicyRaspberryPie/config.yml with contents &uarr;

```yaml
api_port: 4711
console-command-whitelist:
  - help.*
```

### CODE REVIEW

**Review by Wolf2323 — CHANGES_REQUESTED** [review]
Performe command should be on the console, not on the player

**Comment by inkarkat** [comment]
> Performe command should be on the console, not on the player

But the console basically has admin rights, doesn't it? That would then require something like a whitelist to avoid that students are able to execute arbitrary commands on the server.

**Comment by Wolf2323** [comment]
But we want to execute commands that the player has no right to, as far as i know. But yes it should also be an issue then. But with the execution on the player we can also not execute the necessary commands that we need got the game (As far as i know)

**Comment by inkarkat** [comment]
The ticket doesn't say so, and that's also why I cricitized the overly technical nature of it, without referring to the actual game use cases.
We said early on that the game control resides in Minecraft, that's why we don't have a "secure" Python runtime - all Python code is controlled by the student. (Another reason for that decision is allowing us to offer languages other than Python in the future.)

As a consequence, anything coming from Python is inherently untrusted. If we want to execute a command that enables the next level, that decision has to be made in Minecraft; we can only hand out a token (e.g. via sign text) to the Python code so that the Python code can then decide _when_ (but not _if_) to trigger the next level, e.g. by submitting a Minecraft command that returns that token.

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -5,3 +5,4 @@ cmdsvr_port: 4731
 pyexe: "/usr/bin/python3"
 # windows pyexe example
 # pyexe: "C:\\Users\\wensheng\\Anaconda3\\python.exe"
+console-command-whitelist:
```
Comment (Wolf2323) [inline src/main/resources/config.yml:8]: We could discuss here about a default value, even if its empty, but i think it is also ok like it is now

**Review by Wolf2323 — APPROVED** [review]

File: src/main/resources/config.yml (hunk 1 above)
Comment (inkarkat) [reply src/main/resources/config.yml:8]: That would be an empty list... seems I can use JSON syntax (`[]`) in YAML for that.

**Pull request merged**